test.js

```diff
@@ -436,3 +436,447 @@ test('highlight()', function (t) {
+      )
+      .toString(),
+    [
+      '<pre><code class="hljs lang-latex"><span class="hljs-keyword">\\begin</span>{document}',
+      '<span class="hljs-keyword">\\end</span>{document}</code></pre>'
+    ].join('\n'),
+    'should parse custom language'
+  )
+
+  t.equal(
+    rehype()
+      .data('settings', {fragment: true})
+      .use(light)
+      .processSync(
+        [
+          '<h1>Hello World!</h1>',
+          '',
+          '<pre><code class="hljs lang-js"><span class="hljs-keyword">var</span> name = <span class="hljs-string">"World"</span>;',
+          '<span class="hljs-built_in">console</span>.log(<span class="hljs-string">"Hello, "</span> + name + <span class="hljs-string">"!"</span>)</code></pre>'
+        ].join('\n')
+      )
+      .toString(),
+    [
+      '<h1>Hello World!</h1>',
+      '',
+      '<pre><code class="hljs lang-js"><span class="hljs-keyword">var</span> name = <span class="hljs-string">"World"</span>;',
+      '<span class="hljs-built_in">console</span>.log(<span class="hljs-string">"Hello, "</span> + name + <span class="hljs-string">"!"</span>)</code></pre>'
+    ].join('\n'),
+    'should reprocess exact'
+  )
+
+  t.equal(
+    rehype()
+      .data('settings', {fragment: true})
+      .use(light)
+      .processSync(
+        [
+          '<h1>Hello World!</h1>',
+          '',
+          '<pre><code><!--TODO-->"use strict";</code></pre>'
+        ].join('\n')
+      )
+      .toString(),
+    [
+      '<h1>Hello World!</h1>',
+      '',
+      '<pre><code class="hljs language-javascript"><span class="hljs-meta">"use strict"</span>;</code></pre>'
+    ].join('\n'),
+    'should ignore comments'
+  )
+
+  t.equal(
+    rehype()
+      .data('settings', {fragment: true})
+      .use(light)
+      .processSync(
+        [
+          '<h1>Hello World!</h1>',
+          '',
+          '<pre><code>"use strict";<br>console.log("very strict")</code></pre>'
+        ].join('\n')
+      )
+      .toString(),
+    [
+      '<h1>Hello World!</h1>',
+      '',
+      '<pre><code class="hljs language-javascript"><span class="hljs-meta">"use strict"</span>;',
+      '<span class="hljs-built_in">console</span>.log(<span class="hljs-string">"very strict"</span>)</code></pre>'
+    ].join('\n'),
+    'should support `<br>` elements'
+  )
+
+  t.equal(
+    rehype()
+      .data('settings', {fragment: true})
+      .use(light, {
+        languages: {
+          subunit: su,
+          test: function () {
+            return {
+              aliases: ['test'],
+              keywords: {keyword: 'test'}
+            }
+          }
+        }
+      })
+      .processSync(
+        [
+          '<h1>Hello World!</h1>',
+          '',
+          '<pre><code>test normal text</code></pre>'
+        ].join('\n')
+      )
+      .toString(),
+    [
+      '<h1>Hello World!</h1>',
+      '',
+      '<pre><code class="hljs language-subunit"><span class="hljs-keyword">test </span>normal text</code></pre>'
```

Comment:
I'm cheating here! I have no clue why it generates language-subunit for the original test also.

Reply:
wait what? I don‘t understand?

Reply:
When the key `test` is assigned an anonymous function like the one below, it generates `<pre><code class="hljs language-subunit">` and I don't understand how the plugin is getting the `language-subunit` value.

```js
function () {
  return {
    aliases: ['test'],
    keywords: {keyword: 'test'}
  }
}
```

In any case, tonight I'll move the imports up and take out some tests. This test can go away too.

Reply:
Is it not because that test also passes subunit? `subunit: su,`

Reply:
The `subunit: su` is the cheating part I added.

Maybe the easiest way to explain my confusion is: I don't understand the original test `register languages`:
```js

  t.equal(
    rehype()
      .data('settings', {fragment: true})
      .use(highlight, {
        languages: {
          test: function () {
            return {
              aliases: ['test'],
              keywords: {keyword: 'test'}
            }
          }
        }
      })
      .processSync(
        [
          '<h1>Hello World!</h1>',
          '',
          '<pre><code>test normal text</code></pre>'
        ].join('\n')
      )
      .toString(),
    [
      '<h1>Hello World!</h1>',
      '',
      '<pre><code class="hljs language-subunit"><span class="hljs-keyword">test </span>normal text</code></pre>'
    ].join('\n'),
    'shou
```

https://github.com/rehypejs/rehype-highlight/blob/main/test.js#L408

Why is it generating `subunit`?

Reply:
ah, maybe it’s this: highlight.js is global: so registering a language once, means it’s available in other places too 😬